### PR TITLE
refactor(account): select #652

### PR DESCRIPTION
**Pull request by escapedcat**
- switch to TS
- add test

### CODE REVIEW

**Comment by bumi** [comment]
oh no conflicts :(

**Comment by escapedcat** [comment]
> oh no conflicts :(

No worries, it's just a draft. Will fix these and switch to PR when ready.

**Review by bumi — APPROVED** [review]
merging this, but let's quickly chat about the test

File: src/extension/background-script/actions/accounts/__tests__/select.test.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  test("select", async () => {
+    state.getState = jest.fn().mockReturnValue(mockState);
+
+    const spy = jest.spyOn(mockState, "getConnector");
+
+    expect(await select(message)).toStrictEqual({
+      data: { unlocked: true },
+    });
+
```
[the remark this answers is not in the document]
Comment (escapedcat) [reply src/extension/background-script/actions/accounts/__tests__/select.test.ts:50]: You men a instead of mocking the connector?

**Pull request merged**